web/src/app/components/cdf/cdf.component.ts

```diff
@@ -0,0 +1,212 @@
+import { Component, ElementRef, Input, ViewChild, AfterViewInit } from '@angular/core';
+import * as d3 from 'd3';
+
+import { step189_2020 } from '../../../proto/step189_2020';
+
+interface CdfData {
+  duration: number; // Minutes between completed stage and first non-empty stage
+  probability: number; // Rank of the duration divided by number of points
+}
+
+type d3SVG = d3.Selection<SVGSVGElement, undefined, null, undefined>;
```

Comment:
Would be good to explain in a comment what this type is doing and why we need it.